pkg/p2p/kadcast/packet.go

```diff
@@ -0,0 +1,252 @@
+package kadcast
+
+import (
+	"encoding/binary"
+	"log"
+	"net"
+
+	"github.com/dusk-network/dusk-blockchain/pkg/util/container/ring"
+)
+
+// Packet represents a Kadcast packet which is
+// the payload of the TCP or UDP packet received.
+type Packet struct {
+	headers [24]byte
+	payload []byte
+}
+
+// Builds a `Packet` from the headers and the payload.
+func makePacket(headers [24]byte, payload []byte) Packet {
+	return Packet{
+		headers: headers,
+		payload: payload,
+	}
+}
+
+// -------- General Packet De/Serialization tools -------- //
+
+// Gets a stream of bytes and slices it between headers of Kadcast
+// protocol and the payload.
+func getPacketFromStream(stream []byte) Packet {
+	var headers [24]byte
+	copy(headers[:], stream[0:23])
+	return Packet{
+		headers: headers,
+		payload: stream[23:],
+	}
+}
+
+// Deserializes the packet into an slice of bytes.
+func (pac Packet) asBytes() []byte {
+	return append(pac.headers[:], pac.payload[:]...)
```

Comment:
Is it faster or something like this? Since adds a lot of verbosity.

Changed, just curious...

Reply:
`append` and `variadics` is known to escape to heap. Since you are using arrays already for `headers`, you are implicitly prioritizing performance over verbosity/flexibility. Therefore, using fixed size slices is more consistent with the approach you chose